service/src/main/java/uk/gov/hmcts/reform/fpl/service/UploadDocumentsService.java

```diff
@@ -0,0 +1,124 @@
+
+        Document socialWorkStatementDocument = setUpdatedByAndDateAndTimeOnDocuments(
+            caseData.getSocialWorkStatementDocument(), caseDataBefore.getSocialWorkStatementDocument());
+
+        Document socialWorkAssessmentDocument = setUpdatedByAndDateAndTimeOnDocuments(
+            caseData.getSocialWorkAssessmentDocument(), caseDataBefore.getSocialWorkAssessmentDocument());
+
+        Document socialWorkCarePlanDocument = setUpdatedByAndDateAndTimeOnDocuments(
+            caseData.getSocialWorkCarePlanDocument(), caseDataBefore.getSocialWorkCarePlanDocument());
+
+        Document socialWorkEvidenceTemplateDocument = setUpdatedByAndDateAndTimeOnDocuments(
+            caseData.getSocialWorkEvidenceTemplateDocument(), caseDataBefore.getSocialWorkEvidenceTemplateDocument());
+
+        Document thresholdDocument = setUpdatedByAndDateAndTimeOnDocuments(
+            caseData.getThresholdDocument(), caseDataBefore.getThresholdDocument());
+
+        Document checklistDocument = setUpdatedByAndDateAndTimeOnDocuments(
+            caseData.getChecklistDocument(), caseDataBefore.getChecklistDocument());
+
+        CourtBundle courtBundleDocument = setUpdatedByAndDateAndTimeOnDocuments(
+            caseData.getCourtBundle(), caseDataBefore.getCourtBundle());
+
+        Map<String, Object> updatedCaseData = new HashMap<>();
+
+        updatedCaseData.put("documents_socialWorkOther", otherSocialWorkDocuments);
+        updatedCaseData.put("documents_socialWorkChronology_document", socialWorkChronologyDocument);
+        updatedCaseData.put("documents_socialWorkStatement_document", socialWorkStatementDocument);
+        updatedCaseData.put("documents_socialWorkAssessment_document", socialWorkAssessmentDocument);
+        updatedCaseData.put("documents_socialWorkCarePlan_document", socialWorkCarePlanDocument);
+        updatedCaseData.put("documents_socialWorkEvidenceTemplate_document", socialWorkEvidenceTemplateDocument);
+        updatedCaseData.put("documents_threshold_document", thresholdDocument);
+        updatedCaseData.put("documents_checklist_document", checklistDocument);
+        updatedCaseData.put("courtBundle", courtBundleDocument);
+
+        return updatedCaseData;
+    }
+
+    public <T extends DocumentMetaData> List<Element<T>> setUpdatedByAndDateAndTimeOnDocuments(
+        List<Element<T>> currentDocuments,
+        List<Element<T>> previousDocuments) {
+
+        Predicate<Element<T>> doesNotContainInOldDocs =
+            doc -> previousDocuments != null && !previousDocuments.contains(doc);
+
+        currentDocuments.stream()
+            .filter(doesNotContainInOldDocs)
+            .forEach(doc -> findElement(doc.getId(), previousDocuments)
+                .ifPresent(e -> {
+                    if (!e.getValue().getTypeOfDocument().equals(doc.getValue().getTypeOfDocument())) {
+                        setUpdatedByAndDateTime(doc);
```

Comment:
I've a feeling this can throw a NPE exception if documents are included in the SWET.

Reply:
Tested and confirmed not the case.

Reply:
Tested it, all seems fine